src/web3/Client.ts

```diff
@@ -26,15 +33,30 @@ export class Client implements IClient {
    */
   public constructor(
     private clientConfig: IClientConfig,
-    baseAccount?: IAccount,
+    baseAccount?: IAccount | IAccountWalletProvider,
```

Comment:
Why not using a IBaseAccount instead and let people instantiate Client class using meaningful helper such as:
const client =  newBrowserWalletProvider() or const client = newNodeJSProvider() ?
this abviously imply to define the function newBrowserWalletProvider and newNodeJSProvider

Reply:
Done